apple/Sources/FerrostarCore/FerrostarCore.swift

```diff
@@ -126,7 +126,9 @@ public protocol FerrostarCoreDelegate: AnyObject {
         observableState = FerrostarObservableState(snappedLocation: location, heading: locationProvider.lastHeading, fullRoute: route.geometry, steps: route.inner.steps)
         let controller = NavigationController(route: route.inner, config: NavigationControllerConfig(stepAdvance: stepAdvance.ffiValue))
         navigationController = controller
-        update(newState: controller.getInitialState(location: location.userLocation), location: location)
+        DispatchQueue.main.async {
```

Comment:
It looks like you addressed this y making the entire `update` method do an async dispatch to the main thread, but I'm still not sure why we have to do that 🤔

I have a suspicion that it's because of publishers (and that it now wraps the whole method because it can trigger multiple updates so that's easier) but want to make sure I understand before boing an unconditional dispatch like this. We should probably leave rationale in a comment with a warning to never do anything expensive / blocking in this method.

Reply:
Similar to the old UIKit days, manipulation of UI state still requires ensuring it's done on the main thread. I suspect the reason was that I put some of the ferrostarCore behavior inside a `.task { }` on the NavigationView and it kicked the state from the completion of that. Either way, I think we need a mechanism to ensure the state only gets modified on the mean queue regardless of what queue/task, etc. ferrostarCore is running or called from.

Maybe there's a special feature in the iOS 17+ publisher stuff that automatically ensures that, but I think it's most likely just the `.task { }` e.g. where I'm running get routes.

Reply:
Ok, that makes sense. I'll go look into this again later if I get bored (I seem to remember `@MainActor`), but this is good enough to satisfy me for now 😂